.github/workflows/auto-set-base.yml

```diff
@@ -0,0 +1,23 @@
+name: Auto-set Base Branch to Develop
+
+on:
+  pull_request:
+    types: [opened]
+
+jobs:
+  set-base-branch:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Check and Update Base Branch
+        run: |
+          if [ "${{ github.event.pull_request.base.ref }}" != "develop" ]; then
+            gh pr edit ${{ github.event.pull_request.number }} --base develop
+            echo "Base branch updated to 'develop'.";
+          else
+            echo "Base branch is already 'develop'.";
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Not a common case, but I think this should only happen if the target branch is `main`